core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -140,6 +141,32 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
         self ! delayedPollMsg
       }
 
+    case SubscribeWithStartTimestamp(timestamp, topics, listener) =>
+      scheduleFirstPollTask()
+      consumer.subscribe(topics.toList.asJava, new WrappedAutoPausedListener(consumer, listener))
+      // -- Seek offset
+      consumer.poll(0)
+      val assignedTopicPartitions = consumer.assignment()
+
+      // Get offset for a given timestamp for each topic/partition couple
+      val timestampsToSearch = assignedTopicPartitions.asScala.map(
+        partition =>
+          (new TopicPartition(partition.topic(), partition.partition()), timestamp)
+      ).toMap
+
+      assignedTopicPartitions.asScala.map(
```

Comment:
Since you're not interested in the result, but want to perform a side effect for each element, `foreach` would be more idiomatic